### PR TITLE
488 migrate to vipps recurring v3

### PR DESCRIPTION
**Pull request by philiphand**
Migrates Vipps recurring endpoints to work with the new V3 Recurring API

#488 

---

Tested on devices

- [ ] Desktop 💻
- [ ] Mobile 📱

Tests

- [ ] All tests are running ✔️
- [ ] Test are updated 🧪
- [ ] Code Review 👩‍💻
- [ ] QA 👌

⏲️ Time spent on CR:

⏲️ Time spent on QA:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ env.list
 dist/
 .nyc_output/
 .env
+.env.local
```
Comment (adamalfredsson) [inline .gitignore:10]: I think dotenv does not load `.env.local` by default, only `.env`

File: src/custom_modules/vipps.ts [hunk 2]
```diff
@@ -201,7 +201,7 @@ module.exports = {
     };
 
     let initiateRequest = await request.post({
-      uri: `https://${config.vipps_api_url}/ecomm/v2/payments`,
+      uri: `https://${config.vipps_api_url}/ecomm/v3/payments`,
```
Comment (adamalfredsson) [inline src/custom_modules/vipps.ts:204]: I think ecomm v3 API [might not exist](https://developer.vippsmobilepay.com/docs/APIs/ecom-api/vipps-ecom-api/)

File: src/custom_modules/vipps.ts [hunk 3]
```diff
@@ -557,13 +557,18 @@ module.exports = {
...
-      intervalCount: 1,
+      pricing: {
+        type: "LEGACY",
+        amount: realAmount,
+        currency: "NOK",
+      },
+      interval: {
+        unit: "MONTH",
+        count: 1,
+      },
```
Comment (adamalfredsson) [inline src/custom_modules/vipps.ts:568]: Looks solid!

**Review by adamalfredsson — CHANGES_REQUESTED** [review]

File: src/custom_modules/vipps.ts (hunk 2 above)
Comment (philiphand) [reply src/custom_modules/vipps.ts:204]: Ohh good catch! I will change that!

**Review by adamalfredsson — APPROVED** [review]
🎉

**Pull request merged**